daemon/cluster/convert/service.go

```diff
@@ -272,7 +272,7 @@ func restartPolicyToGRPC(p *types.RestartPolicy) (*swarmapi.RestartPolicy, error
 	var rp *swarmapi.RestartPolicy
 	if p != nil {
 		rp = &swarmapi.RestartPolicy{}
-		sanatizedCondition := strings.ToUpper(strings.Replace(string(p.Condition), "-", "_", -1))
+		sanatizedCondition := strings.ToUpper(p.Condition.String())
```

Comment:
Perhaps it should be converted back in `restartPolicyFromGRPC`; https://github.com/docker/docker/pull/27014/files#diff-69edea08b760e2a4f8eb76a11e7f2e89R256